### PR TITLE
Fixed memory leak in search fragment

### LINKED ISSUE #3148: MemoryLeak on `SearchFragement`
**Steps to reproduce the behavior:**
1. Open any zim files
2. Search any keywords

**Screenshots**

**Environment**
- Version of Kiwix Android : 3.6.0
- Device : realme
- OS version : 11

<details><summary> Logs  </summary>

```
2022-12-08 17:10:58.649  4913-4913  ViewRootIm...rActivity] org.kiwix.kiwixmobile                D  processMotionEvent MotionEvent { action=ACTION_UP, actionButton=0, id[0]=0, x[0]=516.0, y[0]=861.0, toolType[0]=TOOL_TYPE_FINGER, buttonState=0, classification=NONE, metaState=0, flags=0x0, edgeFlags=0x0, pointerCount=1, historySize=0, eventTime=89054745, downTime=89054647, deviceId=3, source=0x1002, displayId=0 }
2022-12-08 17:10:58.683  4913-4913  LeakCanary              org.kiwix.kiwixmobile                D  ​
2022-12-08 17:10:58.684  4913-4913  LeakCanary              org.kiwix.kiwixmobile                D  ┬───
2022-12-08 17:10:58.684  4913-4913  LeakCanary              org.kiwix.kiwixmobile                D  │ GC Root: System class
2022-12-08 17:10:58.684  4913-4913  LeakCanary              org.kiwix.kiwixmobile                D  │
2022-12-08 17:10:58.684  4913-4913  LeakCanary              org.kiwix.kiwixmobile                D  ├─ android.app.ActivityThread class
2022-12-08 17:10:58.684  4913-4913  LeakCanary              org.kiwix.kiwixmobile                D  │    Leaking: NO (KiwixMainActivity↓ is not leaking and a class is never leaking)
2022-12-08 17:10:58.684  4913-4913  LeakCanary              org.kiwix.kiwixmobile                D  │    ↓ static ActivityThread.sCurrentActivityThread
2022-12-08 17:10:58.684  4913-4913  LeakCanary              org.kiwix.kiwixmobile                D  ├─ android.app.ActivityThread instance
2022-12-08 17:10:58.684  4913-4913  LeakCanary              org.kiwix.kiwixmobile                D  │    Leaking: NO (KiwixMainActivity↓ is not leaking)
2022-12-08 17:10:58.684  4913-4913  LeakCanary              org.kiwix.kiwixmobile                D  │    mInitialApplication instance of org.kiwix.kiwixmobile.KiwixApp
2022-12-08 17:10:58.684  4913-4913  LeakCanary              org.kiwix.kiwixmobile                D  │    mSystemContext instance of android.app.ContextImpl
2022-12-08 17:10:58.684  4913-4913  LeakCanary              org.kiwix.kiwixmobile                D  │    mSystemUiContext instance of android.app.ContextImpl
2022-12-08 17:10:58.684  4913-4913  LeakCanary              org.kiwix.kiwixmobile                D  │    ↓ ActivityThread.mActivities
2022-12-08 17:10:58.684  4913-4913  LeakCanary              org.kiwix.kiwixmobile                D  ├─ android.util.ArrayMap instance
2022-12-08 17:10:58.684  4913-4913  LeakCanary              org.kiwix.kiwixmobile                D  │    Leaking: NO (KiwixMainActivity↓ is not leaking)
2022-12-08 17:10:58.684  4913-4913  LeakCanary              org.kiwix.kiwixmobile                D  │    ↓ ArrayMap.mArray
2022-12-08 17:10:58.684  4913-4913  LeakCanary              org.kiwix.kiwixmobile                D  ├─ java.lang.Object[] array
2022-12-08 17:10:58.684  4913-4913  LeakCanary              org.kiwix.kiwixmobile                D  │    Leaking: NO (KiwixMainActivity↓ is not leaking)
2022-12-08 17:10:58.684  4913-4913  LeakCanary              org.kiwix.kiwixmobile                D  │    ↓ Object[1]
2022-12-08 17:10:58.684  4913-4913  LeakCanary              org.kiwix.kiwixmobile                D  ├─ android.app.ActivityThread$ActivityClientRecord instance
2022-12-08 17:10:58.684  4913-4913  LeakCanary              org.kiwix.kiwixmobile                D  │    Leaking: NO (KiwixMainActivity↓ is not leaking)
2022-12-08 17:10:58.684  4913-4913  LeakCanary              org.kiwix.kiwixmobile                D  │    activity instance of org.kiwix.kiwixmobile.main.KiwixMainActivity with mDestroyed = false
2022-12-08 17:10:58.684  4913-4913  LeakCanary              org.kiwix.kiwixmobile                D  │    ↓ ActivityThread$ActivityClientRecord.activity
2022-12-08 17:10:58.684  4913-4913  LeakCanary              org.kiwix.kiwixmobile                D  ├─ org.kiwix.kiwixmobile.main.KiwixMainActivity instance
2022-12-08 17:10:58.684  4913-4913  LeakCanary              org.kiwix.kiwixmobile                D  │    Leaking: NO (NavigationHostFragment↓ is not leaking and Activity#mDestroyed is false)
2022-12-08 17:10:58.684  4913-4913  LeakCanary              org.kiwix.kiwixmobile                D  │    mApplication instance of org.kiwix.kiwixmobile.KiwixApp
2022-12-08 17:10:58.684  4913-4913  LeakCanary              org.kiwix.kiwixmobile                D  │    mBase instance of androidx.appcompat.view.ContextThemeWrapper
2022-12-08 17:10:58.684  4913-4913  LeakCanary              org.kiwix.kiwixmobile                D  │    ↓ KiwixMainActivity.navController$delegate
2022-12-08 17:10:58.684  4913-4913  LeakCanary              org.kiwix.kiwixmobile                D  ├─ kotlin.SynchronizedLazyImpl instance
2022-12-08 17:10:58.684  4913-4913  LeakCanary              org.kiwix.kiwixmobile                D  │    Leaking: NO (NavigationHostFragment↓ is not leaking)
2022-12-08 17:10:58.684  4913-4913  LeakCanary              org.kiwix.kiwixmobile                D  │    ↓ SynchronizedLazyImpl._value
2022-12-08 17:10:58.684  4913-4913  LeakCanary              org.kiwix.kiwixmobile                D  ├─ androidx.navigation.NavHostController instance
2022-12-08 17:10:58.684  4913-4913  LeakCanary              org.kiwix.kiwixmobile                D  │    Leaking: NO (NavigationHostFragment↓ is not leaking)
2022-12-08 17:10:58.684  4913-4913  LeakCanary              org.kiwix.kiwixmobile                D  │    activity instance of org.kiwix.kiwixmobile.main.KiwixMainActivity with mDestroyed = false
2022-12-08 17:10:58.684  4913-4913  LeakCanary              org.kiwix.kiwixmobile                D  │    context instance of org.kiwix.kiwixmobile.main.KiwixMainActivity with mDestroyed = false
2022-12-08 17:10:58.684  4913-4913  LeakCanary              org.kiwix.kiwixmobile                D  │    ↓ NavController.lifecycleOwner
2022-12-08 17:10:58.684  4913-4913  LeakCanary              org.kiwix.kiwixmobile                D  ├─ org.kiwix.kiwixmobile.core.main.NavigationHostFragment instance
2022-12-08 17:10:58.684  4913-4913  LeakCanary              org.kiwix.kiwixmobile                D  │    Leaking: NO (SearchFragment↓ is not leaking and Fragment#mFragmentManager is not null)
2022-12-08 17:10:58.684  4913-4913  LeakCanary              org.kiwix.kiwixmobile                D  │    ↓ Fragment.mChildFragmentManager
2022-12-08 17:10:58.684  4913-4913  LeakCanary              org.kiwix.kiwixmobile                D  ├─ androidx.fragment.app.FragmentManagerImpl instance
2022-12-08 17:10:58.684  4913-4913  LeakCanary              org.kiwix.kiwixmobile                D  │    Leaking: NO (SearchFragment↓ is not leaking)
2022-12-08 17:10:58.684  4913-4913  LeakCanary              org.kiwix.kiwixmobile                D  │    ↓ FragmentManager.mFragmentStore
2022-12-08 17:10:58.684  4913-4913  LeakCanary              org.kiwix.kiwixmobile                D  ├─ androidx.fragment.app.FragmentStore instance
2022-12-08 17:10:58.684  4913-4913  LeakCanary              org.kiwix.kiwixmobile                D  │    Leaking: NO (SearchFragment↓ is not leaking)
2022-12-08 17:10:58.684  4913-4913  LeakCanary              org.kiwix.kiwixmobile                D  │    ↓ FragmentStore.mActive
2022-12-08 17:10:58.684  4913-4913  LeakCanary              org.kiwix.kiwixmobile                D  ├─ java.util.HashMap instance
2022-12-08 17:10:58.684  4913-4913  LeakCanary              org.kiwix.kiwixmobile                D  │    Leaking: NO (SearchFragment↓ is not leaking)
2022-12-08 17:10:58.684  4913-4913  LeakCanary              org.kiwix.kiwixmobile                D  │    ↓ HashMap["b780a08f-28e7-4961-b740-c2974aaa62c6"]
2022-12-08 17:10:58.684  4913-4913  LeakCanary              org.kiwix.kiwixmobile                D  ├─ androidx.fragment.app.FragmentStateManager instance
2022-12-08 17:10:58.684  4913-4913  LeakCanary              org.kiwix.kiwixmobile                D  │    Leaking: NO (SearchFragment↓ is not leaking)
2022-12-08 17:10:58.684  4913-4913  LeakCanary              org.kiwix.kiwixmobile                D  │    ↓ FragmentStateManager.mFragment
2022-12-08 17:10:58.684  4913-4913  LeakCanary              org.kiwix.kiwixmobile                D  ├─ org.kiwix.kiwixmobile.core.search.SearchFragment instance
2022-12-08 17:10:58.684  4913-4913  LeakCanary              org.kiwix.kiwixmobile                D  │    Leaking: NO (Fragment#mFragmentManager is not null)
2022-12-08 17:10:58.684  4913-4913  LeakCanary              org.kiwix.kiwixmobile                D  │    ↓ SearchFragment.searchView
2022-12-08 17:10:58.684  4913-4913  LeakCanary              org.kiwix.kiwixmobile                D  │                     ~~~~~~~~~~
2022-12-08 17:10:58.684  4913-4913  LeakCanary              org.kiwix.kiwixmobile                D  ├─ androidx.appcompat.widget.SearchView instance
2022-12-08 17:10:58.685  4913-4913  LeakCanary              org.kiwix.kiwixmobile                D  │    Leaking: UNKNOWN
2022-12-08 17:10:58.685  4913-4913  LeakCanary              org.kiwix.kiwixmobile                D  │    Retaining 30.0 kB in 572 objects
2022-12-08 17:10:58.685  4913-4913  LeakCanary              org.kiwix.kiwixmobile                D  │    View not part of a window view hierarchy
2022-12-08 17:10:58.685  4913-4913  LeakCanary              org.kiwix.kiwixmobile                D  │    View.mAttachInfo is null (view detached)
2022-12-08 17:10:58.685  4913-4913  LeakCanary              org.kiwix.kiwixmobile                D  │    View.mID = R.id.menu_search
2022-12-08 17:10:58.685  4913-4913  LeakCanary              org.kiwix.kiwixmobile                D  │    View.mWindowAttachCount = 1
2022-12-08 17:10:58.685  4913-4913  LeakCanary              org.kiwix.kiwixmobile                D  │    mContext instance of androidx.appcompat.view.ContextThemeWrapper, wrapping activity org.kiwix.kiwixmobile.main.
2022-12-08 17:10:58.685  4913-4913  LeakCanary              org.kiwix.kiwixmobile                D  │    KiwixMainActivity with mDestroyed = false
2022-12-08 17:10:58.685  4913-4913  LeakCanary              org.kiwix.kiwixmobile                D  │    ↓ View.mParent
2022-12-08 17:10:58.685  4913-4913  LeakCanary              org.kiwix.kiwixmobile                D  │           ~~~~~~~
2022-12-08 17:10:58.685  4913-4913  LeakCanary              org.kiwix.kiwixmobile                D  ├─ androidx.appcompat.widget.Toolbar instance
2022-12-08 17:10:58.685  4913-4913  LeakCanary              org.kiwix.kiwixmobile                D  │    Leaking: UNKNOWN
2022-12-08 17:10:58.685  4913-4913  LeakCanary              org.kiwix.kiwixmobile                D  │    Retaining 8.3 kB in 158 objects
2022-12-08 17:10:58.685  4913-4913  LeakCanary              org.kiwix.kiwixmobile                D  │    View not part of a window view hierarchy
2022-12-08 17:10:58.685  4913-4913  LeakCanary              org.kiwix.kiwixmobile                D  │    View.mAttachInfo is null (view detached)
2022-12-08 17:10:58.685  4913-4913  LeakCanary              org.kiwix.kiwixmobile                D  │    View.mID = R.id.toolbar
2022-12-08 17:10:58.685  4913-4913  LeakCanary              org.kiwix.kiwixmobile                D  │    View.mWindowAttachCount = 1
2022-12-08 17:10:58.685  4913-4913  LeakCanary              org.kiwix.kiwixmobile                D  │    mPopupContext instance of android.view.ContextThemeWrapper, wrapping activity org.kiwix.kiwixmobile.main.
2022-12-08 17:10:58.685  4913-4913  LeakCanary              org.kiwix.kiwixmobile                D  │    KiwixMainActivity with mDestroyed = false
2022-12-08 17:10:58.685  4913-4913  LeakCanary              org.kiwix.kiwixmobile                D  │    mContext instance of androidx.appcompat.view.ContextThemeWrapper, wrapping activity org.kiwix.kiwixmobile.main.
2022-12-08 17:10:58.685  4913-4913  LeakCanary              org.kiwix.kiwixmobile                D  │    KiwixMainActivity with mDestroyed = false
2022-12-08 17:10:58.685  4913-4913  LeakCanary              org.kiwix.kiwixmobile                D  │    ↓ View.mParent
2022-12-08 17:10:58.685  4913-4913  LeakCanary              org.kiwix.kiwixmobile                D  │           ~~~~~~~
2022-12-08 17:10:58.685  4913-4913  LeakCanary              org.kiwix.kiwixmobile                D  ├─ android.widget.LinearLayout instance
2022-12-08 17:10:58.685  4913-4913  LeakCanary              org.kiwix.kiwixmobile                D  │    Leaking: UNKNOWN
2022-12-08 17:10:58.685  4913-4913  LeakCanary              org.kiwix.kiwixmobile                D  │    Retaining 12.4 kB in 272 objects
2022-12-08 17:10:58.685  4913-4913  LeakCanary              org.kiwix.kiwixmobile                D  │    View not part of a window view hierarchy
2022-12-08 17:10:58.685  4913-4913  LeakCanary              org.kiwix.kiwixmobile                D  │    View.mAttachInfo is null (view detached)
2022-12-08 17:10:58.685  4913-4913  LeakCanary              org.kiwix.kiwixmobile                D  │    View.mWindowAttachCount = 1
2022-12-08 17:10:58.685  4913-4913  LeakCanary              org.kiwix.kiwixmobile                D  │    mContext instance of org.kiwix.kiwixmobile.main.KiwixMainActivity with mDestroyed = false
2022-12-08 17:10:58.685  4913-4913  LeakCanary              org.kiwix.kiwixmobile                D  │    ↓ View.mParent
2022-12-08 17:10:58.685  4913-4913  LeakCanary              org.kiwix.kiwixmobile                D  │           ~~~~~~~
2022-12-08 17:10:58.685  4913-4913  LeakCanary              org.kiwix.kiwixmobile                D  ╰→ androidx.coordinatorlayout.widget.CoordinatorLayout instance
2022-12-08 17:10:58.685  4913-4913  LeakCanary              org.kiwix.kiwixmobile                D  ​     Leaking: YES (ObjectWatcher was watching this because org.kiwix.kiwixmobile.core.search.SearchFragment received
2022-12-08 17:10:58.685  4913-4913  LeakCanary              org.kiwix.kiwixmobile                D  ​     Fragment#onDestroyView() callback (references to its views should be cleared to prevent leaks))
2022-12-08 17:10:58.685  4913-4913  LeakCanary              org.kiwix.kiwixmobile                D  ​     Retaining 11.1 kB in 255 objects
2022-12-08 17:10:58.685  4913-4913  LeakCanary              org.kiwix.kiwixmobile                D  ​     key = 58cc0592-7932-40b7-858a-506e2bd8f2a6
2022-12-08 17:10:58.685  4913-4913  LeakCanary              org.kiwix.kiwixmobile                D  ​     watchDurationMillis = 334397
2022-12-08 17:10:58.685  4913-4913  LeakCanary              org.kiwix.kiwixmobile                D  ​     retainedDurationMillis = 329395
2022-12-08 17:10:58.685  4913-4913  LeakCanary              org.kiwix.kiwixmobile                D  ​     View not part of a window view hierarchy
2022-12-08 17:10:58.685  4913-4913  LeakCanary              org.kiwix.kiwixmobile                D  ​     View.mAttachInfo is null (view detached)
2022-12-08 17:10:58.685  4913-4913  LeakCanary              org.kiwix.kiwixmobile                D  ​     View.mWindowAttachCount = 1
2022-12-08 17:10:58.685  4913-4913  LeakCanary              org.kiwix.kiwixmobile                D  ​     mContext instance of org.kiwix.kiwixmobile.main.KiwixMainActivity with mDestroyed = false
2022-12-08 17:10:58.685  4913-4913  LeakCanary              org.kiwix.kiwixmobile                D  
2022-12-08 17:10:58.685  4913-4913  LeakCanary              org.kiwix.kiwixmobile                D  METADATA
2022-12-08 17:10:58.685  4913-4913  LeakCanary              org.kiwix.kiwixmobile                D  
2022-12-08 17:10:58.685  4913-4913  LeakCanary              org.kiwix.kiwixmobile                D  Build.VERSION.SDK_INT: 30
2022-12-08 17:10:58.685  4913-4913  LeakCanary              org.kiwix.kiwixmobile                D  Build.MANUFACTURER: realme
2022-12-08 17:10:58.685  4913-4913  LeakCanary              org.kiwix.kiwixmobile                D  LeakCanary version: 2.9.1
2022-12-08 17:10:58.685  4913-4913  LeakCanary              org.kiwix.kiwixmobile                D  App process name: org.kiwix.kiwixmobile
2022-12-08 17:10:58.685  4913-4913  LeakCanary              org.kiwix.kiwixmobile                D  Class count: 23925
2022-12-08 17:10:58.685  4913-4913  LeakCanary              org.kiwix.kiwixmobile                D  Instance count: 191647
2022-12-08 17:10:58.685  4913-4913  LeakCanary              org.kiwix.kiwixmobile                D  Primitive array count: 120292
2022-12-08 17:10:58.685  4913-4913  LeakCanary              org.kiwix.kiwixmobile                D  Object array count: 28939
2022-12-08 17:10:58.685  4913-4913  LeakCanary              org.kiwix.kiwixmobile                D  Thread count: 62
2022-12-08 17:10:58.685  4913-4913  LeakCanary              org.kiwix.kiwixmobile                D  Heap total bytes: 27314624
2022-12-08 17:10:58.685  4913-4913  LeakCanary              org.kiwix.kiwixmobile                D  Bitmap count: 8
2022-12-08 17:10:58.685  4913-4913  LeakCanary              org.kiwix.kiwixmobile                D  Bitmap total bytes: 31052168
2022-12-08 17:10:58.685  4913-4913  LeakCanary              org.kiwix.kiwixmobile                D  Large bitmap count: 0
2022-12-08 17:10:58.685  4913-4913  LeakCanary              org.kiwix.kiwixmobile                D  Large bitmap total bytes: 0

```
</details>

### PR DESCRIPTION
**Pull request by MohitMaliFtechiz**
Fixes #3148

### CODE REVIEW

File: core/src/main/java/org/kiwix/kiwixmobile/core/search/SearchFragment.kt [hunk 1]
```diff
@@ -100,6 +93,12 @@ class SearchFragment : BaseFragment() {
 
   override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
     super.onViewCreated(view, savedInstanceState)
+    searchAdapter = SearchAdapter(
+      RecentSearchDelegate(::onItemClick, ::onItemClickNewTab) {
+        searchViewModel.actions.offer(OnItemLongClick(it))
+      },
+      ZimSearchResultDelegate(::onItemClick, ::onItemClickNewTab)
+    )
```
Comment (gouri-panda) [inline core/src/main/java/org/kiwix/kiwixmobile/core/search/SearchFragment.kt:101]: We should create `searchAdapter` lazily.

**Comment by gouri-panda** [comment]
@MohitMaliDeveloper Just a small change then we are good to go!

File: core/src/main/java/org/kiwix/kiwixmobile/core/search/SearchFragment.kt (hunk 1 above)
Comment (gouri-panda) [reply core/src/main/java/org/kiwix/kiwixmobile/core/search/SearchFragment.kt:101]: Ok @MohitMaliDeveloper . There is one solution to hack [this](https://stackoverflow.com/questions/50173899/null-property-when-accessing-lazy-non-nullable-property-of-android-sensor). But I think it's too much engineering . What do you think?

**Review by gouri-panda — APPROVED** [review]
We should merge this after #3155 .

**Comment by MohitMaliFtechiz** [comment]
I am closing this PR , all this changes are placed on https://github.com/kiwix/kiwix-android/pull/3158

**Pull request closed without merging**